pkg/services/validator/azure.go

```diff
@@ -347,8 +335,100 @@ func configureAzureRBACRulePermissionSets(r *plug.RBACRule) error {
 	}
 }
 
+// configureCommunityGalleryImageRules sets up zero or more Community Gallery Image rules based on
+// pre-existing files or user input.
+func configureCommunityGalleryImageRules(c *components.AzurePluginConfig, ruleNames *[]string) error {
+	log.InfoCLI(`
+	Community gallery image validation rules ensure that images are publicly available via community galleries.
+	`)
+
+	validateCommunityGalleryImage, err := prompts.ReadBool("Enable Community Gallery Image validation", true)
+	if err != nil {
+		return err
+	}
+	if !validateCommunityGalleryImage {
+		c.Validator.CommunityGalleryImageRules = nil
+	}
+	for i, r := range c.Validator.CommunityGalleryImageRules {
+		r := r
+		if err := readCommunityGalleryImageRule(c, &r, i, ruleNames); err != nil {
+			return err
+		}
+	}
+	addRules := true
+	if c.Validator.CommunityGalleryImageRules == nil {
+		c.Validator.CommunityGalleryImageRules = make([]plug.CommunityGalleryImageRule, 0)
+	} else {
+		addRules, err = prompts.ReadBool("Add another Community Gallery Image rule", false)
+		if err != nil {
+			return err
+		}
+	}
+	if !addRules {
+		return nil
+	}
+	for {
+		if err := readCommunityGalleryImageRule(c, &plug.CommunityGalleryImageRule{}, -1, ruleNames); err != nil {
+			return err
+		}
+		add, err := prompts.ReadBool("Add additional Community Gallery Image rule", false)
+		if err != nil {
+			return err
+		}
+		if !add {
+			break
+		}
+	}
+	return nil
+}
+
+// readCommunityGalleryImageRule begins the process of reconfiguring or beginning a new Community
+// Gallery Image rule.
+func readCommunityGalleryImageRule(c *components.AzurePluginConfig, r *plug.CommunityGalleryImageRule, idx int, ruleNames *[]string) error {
+	name := r.Name
+	if name != "" {
```

Comment:
See above comment. Once you refactor to use the `initRule` pattern, this duplication will go away.